### PR TITLE
Adds `BaselineImputer` and Reworks `MarginalImputer`

### LINKED ISSUE #107: Add BaseImputer
for language and vision?

### LINKED ISSUE #261: Add joint marginal imputation method for marginal imputer
Atm. only independent sampling of features is possible. However, in a lot of cases sampling from the joint marginal distribution is necessary to be statistically correct.

We need to extend `MarginalImputer` for this.

### PR DESCRIPTION
**Pull request by mmschlk**
**TLDR**: This PR closes #261 and this PR closes #107.

This pull request introduces several updates to the `shapiq` package, focusing on the addition of a new `BaselineImputer` class and various improvements to existing imputer classes.

### New Features:
* Added `BaselineImputer` class for imputing missing values using predefined baseline values or calculated mean/mode from background data. (`shapiq/games/imputer/baseline_imputer.py`)

### Updates to Existing Features:
* Removed the `sample_replacements` parameter from `MarginalImputer` and added a `joint_marginal_distribution` parameter. A DeprecationWarning for `sample_replacements` has been added. (`shapiq/games/imputer/marginal_imputer.py`)
* Enhanced the `Imputer` base class by adding new attributes (`n_features`, `data`, `model`, `sample_size`) and methods (`fit`, `x` property) to improve flexibility and functionality. (`shapiq/games/imputer/base.py`)

### Codebase Organization:
* Updated imports and `__all__` lists to include the new `BaselineImputer` class. (`shapiq/__init__.py`, `shapiq/games/__init__.py`, `shapiq/games/imputer/__init__.py`)

### Documentation:
* Updated `CHANGELOG.md` to reflect the new features and changes, including the addition of `BaselineImputer` and the removal of the `sample_replacements` parameter from `MarginalImputer`.

### Minor Improvements:
* Added a TODO comment in `shapiq/games/base.py` to consider removing a potentially runtime-increasing check. (`shapiq/games/base.py`)

### CODE REVIEW

File: shapiq/games/imputer/marginal_imputer.py [hunk 1]
```diff
@@ -46,26 +53,25 @@ def __init__(
         sample_replacements: bool = True,
         sample_size: int = 100,
         categorical_features: list[int] = None,
+        joint_marginal_distribution: bool = False,
```
Comment (Advueu963) [inline shapiq/games/imputer/marginal_imputer.py:56]: Maybe adding an small docs for this would be nice.

File: shapiq/games/imputer/base.py [hunk 2]
```diff
@@ -19,30 +19,67 @@ class Imputer(Game):
...
+        sample_size: The number of samples to draw from the background data.
     """
 
     @abstractmethod
     def __init__(
         self,
         model,
         data: np.ndarray,
+        x: Optional[np.ndarray] = None,
+        sample_size: int = 100,
```
Comment (Advueu963) [inline shapiq/games/imputer/base.py:36]: It would be nice to add information about x and sample_size in the docs of __init__.
Could just use the identical comments for BaselineImputer

**Review by Advueu963 — APPROVED** [review]

File: shapiq/games/imputer/marginal_imputer.py (hunk 1 above)
Comment (mmschlk) [reply shapiq/games/imputer/marginal_imputer.py:56]: done

File: shapiq/games/imputer/base.py (hunk 2 above)
Comment (mmschlk) [reply shapiq/games/imputer/base.py:36]: thanks overlooked this!
